src/WebJobs.Script/WebJobs.Script.csproj

```diff
@@ -1,6 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <Import Project="..\..\build\common.props" />
   <PropertyGroup>
+    <TargetFramework>net8.0</TargetFramework>
```

Comment:
The new APIs being used are only available in .NET 7+.

@kshyju I am thinking this needs to be better planned (this is not the only place we'll need to update to use this).

It's fine if want to measure with the changes, but I want to avoid trickling this in.

Reply:
This PR is targeting the feature branch which has net8 TFM.
We could update the code with `#if NET7_0_OR_GREATER` to conditionally use this so that this change can be merged to dev branch. 

@fabiocav there are other places where regex can be replaced with src gen version, but this is the only one I spotted (via cpu profiling using diagnostic tools ) in specialization flow.